editors/sc-ide/widgets/help_browser.cpp

```diff
@@ -120,6 +124,41 @@ HelpBrowser::HelpBrowser( QWidget * parent ):
     setFocusProxy(mWebView);
 }
 
+void HelpBrowser::setUpServer()
+{
+
+    // setup HelpBrowser server
+    QWebSocketServer server("SCIDE HelpBrowser Server", QWebSocketServer::NonSecureMode);
+    for (int i = 0; i < 8; i++) {
+        mServerRunning = server.listen(QHostAddress::LocalHost, mServerPort);
+        if (mServerRunning) {
+            break;
+        }
+        mServerPort += 1;
+    }
```

Comment:
since there are two magic-number constants in this snippet (default value of server port and ergonomic value 8), might be a good idea to pull this out into a separate function, and then you can use `return` instead of `break`, and provide a failure-case return value after the loop, which will clarify the logic. this would be another argument for using only port to indicate the server status, since you can then easily just manage the one variable instead of two:

```cpp
int tryFindOpenServerPort()
{
    int const startPort = 12344;
    QWebSocketServer server(...);
    for (int port = startPort; port < startPort + 8; ++port) {
        if (server.listen(...)) {
            return port;
        }
    }
    return -1; // failure
}
```